edge/pkg/edgehub/clients/wsclient/websocket.go

```diff
@@ -2,127 +2,105 @@ package wsclient
+		ConnUse:          api.UseTypeMessage,
+	}
+	exOpts := api.WSClientOption{Header: make(http.Header)}
+	exOpts.Header.Set("node_id", wsc.config.NodeID)
+	exOpts.Header.Set("project_id", wsc.config.ProjectID)
+	client := &wsclient.Client{Options: option, ExOpts: exOpts}
+
 	for i := 0; i < retryCount; i++ {
-		conn, resp, err := dialer.Dial(wcc.config.URL, wcc.config.ExtendHeader)
+		connection, err := client.Connect()
 		if err != nil {
-			var respMsg string
-			if resp != nil {
-				body, rErr := ioutil.ReadAll(resp.Body)
-				if rErr == nil {
-					respMsg = fmt.Sprintf(", response code: %d, response body: %s", resp.StatusCode, string(body))
-				} else {
-					respMsg = fmt.Sprintf(", response code: %d", resp.StatusCode)
-				}
-				resp.Body.Close()
-			}
-			log.LOGGER.Errorf("Error when init websocket connection%s: %v", respMsg, err)
+			log.LOGGER.Errorf("Init websocket connection failed %s", err.Error())
 		} else {
-			log.LOGGER.Infof("Success to connect Access")
-			wcc.webConn = conn
+			wsc.connection = connection
+			log.LOGGER.Infof("Websocket connect to cloud access successful")
 			return nil
 		}
 		time.Sleep(cloudAccessSleep)
 	}
-	return errors.New("max retry count to connect Access")
+	return errors.New("max retry count reached when connecting to cloud")
 }
 
-//Uninit closes the web socket connection
-func (wcc *WebSocketClient) Uninit() {
-	wcc.webConn.Close()
+//Uninit closes the websocket connection
+func (wsc *WebSocketClient) Uninit() {
+	wsc.connection.Close()
 }
 
-//Send sends the message as binary message through the connection
-func (wcc *WebSocketClient) Send(message model.Message) error {
-	deadline := time.Now().Add(wcc.config.WriteDeadline)
-	wcc.sendLock.Lock()
-	defer wcc.sendLock.Unlock()
-	wcc.webConn.SetWriteDeadline(deadline)
-
-	data, err := json.Marshal(message)
-	if err != nil {
-		return fmt.Errorf("websocket write msg failed with marshal failed. error %s", err.Error())
-	}
-
-	return wcc.webConn.WriteMessage(websocket.BinaryMessage, data)
+//Send sends the message as JSON object through the connection
+func (wsc *WebSocketClient) Send(message model.Message) error {
+	return wsc.connection.WriteMessageAsync(&message)
 }
 
 //Receive reads the binary message through the connection
-func (wcc *WebSocketClient) Receive() (model.Message, error) {
-	var message model.Message
-
-	_, buf, err := wcc.webConn.ReadMessage()
-	if err != nil {
-		return model.Message{}, err
-	}
-
-	err = json.Unmarshal(buf, &message)
-	if err != nil {
-		log.LOGGER.Errorf("Failed to read json: %v", err)
-		return model.Message{}, err
-	}
-
+func (wsc *WebSocketClient) Receive() (model.Message, error) {
+	message := model.Message{}
+	wsc.connection.ReadMessage(&message)
 	return message, nil
 }
 
 //Notify logs info
-func (wcc *WebSocketClient) Notify(authInfo map[string]string) {
-	log.LOGGER.Infof("don't care")
+func (wsc *WebSocketClient) Notify(authInfo map[string]string) {
+	log.LOGGER.Infof("Don not care")
```

Comment:
whats this comment ? can say "no op" instead.

Reply:
@rohitsardesai83 I have updated the code according to your comment. Can you please review again?